### PR TITLE
vsgxchange recipe

### PR DESCRIPTION
**Pull request by psyinf**
**vsgxchange/1.0.2**

First addition to the VulkanSceneGraph (vsg) zoo of additional libraries
---

- [x ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by evilenzo** [comment]
Got such warns when tried to create your package 
```WARN: deprecated: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
WARN: deprecated:     'cpp_info.filenames' used in: vsg/1.0.3, vsgxchange/1.0.2, vulkan-headers/1.3.239.0
WARN: deprecated:     'cpp_info.names' used in: openssl/1.1.1t, vulkan-loader/1.3.239.0, vsg/1.0.3, vsgxchange/1.0.2, libcurl/8.0.0, zlib/1.2.13, vulkan-headers/1.3.239.0
WARN: deprecated:     'cpp_info.build_modules' used in: openssl/1.1.1t
WARN: deprecated:     'env_info' used in: vulkan-loader/1.3.239.0```

**Comment by psyinf** [comment]
@conan-io/barbarians I'm not quite sure why the conan v2.0 pipeline fails with the libcurl here. Any hints?

File: recipes/vsgxchange/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        self.requires("vulkan-loader/1.3.239.0")
+        self.requires("vsg/1.0.3")
+        #optional
+        if self.options.with_curl:
+            self.requires("libcurl/8.0.0")
```
Comment (valgur) [inline recipes/vsgxchange/all/conanfile.py:59]:
```suggestion
            self.requires("libcurl/[>=7.78.0 <9]")
```

File: recipes/vsgxchange/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
```
Comment (valgur) [inline recipes/vsgxchange/all/conanfile.py:79]:
```suggestion
        get(self, **self.conan_data["sources"][self.version], strip_root=True)

```

File: recipes/vsgxchange/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if is_msvc(self):
+            tc.variables["USE_MSVC_RUNTIME_LIBRARY_DLL"] = False
```
Comment (valgur) [inline recipes/vsgxchange/all/conanfile.py:83]: Why is this necessary?

**Review by valgur — COMMENTED** [review]
The MSVC runtime handling could use more info, otherwise LGTM.
